static/js/theme-map/Util/helpers.js

```diff
@@ -1,25 +1,29 @@
+import { GoogleMaps } from '../Maps/Providers/Google.js';
+import { MapboxMaps } from '../Maps/Providers/Mapbox.js';
+
 /**
  * Gets the language locale according to specific fallback logic
  * 1. The user-specified locale to the component
  * 2. If invalid, try using only the first two characters
  * 3. If still invalid, providers fallback to en
  *
  * @param {string} localeStr The user-defined locale string
- * @param {string[]} supportedLocales The locales supported by the current map provider
+ * @param {string} mapProvider name of the current map provider
  * @return {string} The language locale for the map
  */
-const getLanguageForProvider = (localeStr, supportedLocales) => {
+const getLanguageForProvider = (localeStr, mapProvider) => {
   if (localeStr.length == 2) {
     return localeStr;
   } 
 
   if (localeStr.length > 2) {
-    if (supportedLocalesForProvider.includes(localeStr)) {
-      return localeStr;
-    } 
+    const provider = (mapProvider === 'google') ? GoogleMaps : MapboxMaps;
```

Comment:
should we display an error if mapprovider is not google or mapbox? maybe we want to extract this method to a separate helper since it's also used in ThemeMap.js?

Reply:
yeah, that would help if something is not configure right. I will update this

Reply:
do you think it would be better to default to google/mapbox, or just log error and return null?

Reply:
maybe log an error and return 'en'? I think returning null would make it blow up later down the road. If we want it to blow up I think we should just throw an error, though I imagine logging an error is good enough to not hide bugs but still handle things gracefully?

Reply:
ooo sorry you mean for the helper, yeah I think that's fine!

Reply:
we probably don't want to default, I think the provider already defaults to mapbox somewhere else in the code?

Reply:
I think you are referring to the loadAndInitializeMap() in ThemeMap for defaulting to mapbox? which we would replace with our helper call. So I guess to keep the functionality the same, I would default to Mapbox.

Reply:
ah yeah that's my bad